src/environment.ts

```diff
@@ -1,32 +1,23 @@
-import { apply, isUndefined, ObjectCreate, isFunction, hasOwnProperty, ObjectDefineProperty, emptyArray, isArray, map, construct, ESGlobalKeys, getOwnPropertyDescriptor, isExtensible } from './shared';
+import { 
+    apply, 
+    assign, 
+    isUndefined, 
+    ObjectCreate, 
+    isFunction, 
+    hasOwnProperty, 
+    defineProperty, 
+    emptyArray, 
+    isArray, 
+    map, 
+    construct, 
+    ESGlobalKeys, 
+    getOwnPropertyDescriptor, 
+    isExtensible, 
+    setPrototypeOf 
+} from './shared';
 import { SecureProxyHandler } from './secure-proxy-handler';
 import { ReverseProxyHandler } from './reverse-proxy-handler';
 
-/**
- * This method returns a descriptor that given an original setter, and a getter, can use the original
- * getter to return a secure object, but if the sandbox attempts to set it to a new value, this
- * mutation will only affect the sandbox's global object, and the getter will start returning the
- * new provided value rather than calling onto the outer realm. This is to preserve the object graph
- * of the outer realm.
- */
-function getSecureGlobalAccessorDescriptor(env: SecureEnvironment, descriptor: PropertyDescriptor): PropertyDescriptor {
-    const { get: originalGetter } = descriptor;
-    let currentGetter = isUndefined(originalGetter) ? () => undefined : function(this: any): SecureValue {
-        const value: RawValue = apply(originalGetter, env.getRawValue(this), emptyArray);
-        return env.getSecureValue(value);
-    }
-    if (!isUndefined(originalGetter)) {
-        descriptor.get = function get(): SecureValue {
-            return apply(currentGetter, this, emptyArray);
-        };
-    }
-    descriptor.set = function set(v: SecureValue): void {
-        // if a global setter is invoke, the value will be use as it is as the result of the getter operation
-        currentGetter = () => v;
-    };
-    return descriptor;
-}
-
```

Comment:
☝️ Simplified and inlined this because we don't want to mutate passed in descriptors. See further down in the file for it's inlined form.